polaris/hub/external_auth_client.py

```diff
@@ -0,0 +1,138 @@
+import webbrowser
+from typing import Optional
+
+from authlib.common.security import generate_token
+from authlib.integrations.base_client import OAuthError
+from authlib.integrations.httpx_client import OAuth2Client
+from authlib.oauth2 import TokenAuth
+from loguru import logger
+
+from polaris.hub.oauth import ExternalCachedTokenAuth
+from polaris.hub.settings import PolarisHubSettings
+from polaris.utils.errors import PolarisUnauthorizedError
+
+
+class ExternalAuthClient(OAuth2Client):
+    """
+    This authentication client is used to obtain OAuth 2 tokens from Polaris's external OAuth2 server.
+    These can in turn be used to obtain Polaris Hub tokens.
+
+    This class is intended for internal use by the `PolarisHubClient` class, and you should not have to
+    interact with it directly.
```

Comment:
**Suggestion:** Minor, but I like using [admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/) for comments like these so that they stand out more in the generated docs. 

For example:

```suggestion
    Note: Internal use
        This class is intended for internal use by the `PolarisHubClient` class, and you should not have to
        interact with it directly.
```